src/PerfView/PerfViewData.cs

```diff
@@ -5744,6 +5764,27 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                                     }
                                     sumCumMetric += cumMetric;
                                     cumCount++;
+
+                                    string GetAllocationType()
+                                    {
+                                        if (!allocationTypeNames.TryGetValue(callerIndex, out var typeName))
+                                        {
+                                            var cs = data.CallStack();
+                                            while (cs?.CodeAddress.ModuleFile is TraceModuleFile module)
+                                            {
+                                                if (allocationSites.TryGetValue(module, out var heapAllocationSites) &&
+                                                    heapAllocationSites.FirstOrDefault(x => x.Rva == cs.CodeAddress.Address - module.ImageBase)?.TypeName is string name)
+                                                {
+                                                    typeName = name;
```

Comment:
Note that this currently resolves most allocations from STL containers as type "void" because the std::allocator<T> implementation uses void* as the return type. I think the Visual Studio implementation of the feature may special case this to extract the T, but I wanted to keep this simple for a first pass.